utils.ps1

```diff
@@ -0,0 +1,298 @@
+    PowerShell to load DbaTools whenever PowerShell starts.
+.PARAMETER AllHosts
+    By default, this command modifies the CurrentUserCurrentHost profile
+    script.  By specifying the AllHosts switch, the command updates the
+    CurrentUserAllHosts profile.
+.PARAMETER Force
+    Do not check if the specified profile script is already importing
+    DbaTools. Just add Import-Module DbaTools command.
+.PARAMETER StartSshAgent
+    Also add `Start-SshAgent -Quiet` to the specified profile script.
+.EXAMPLE
+    PS C:\> Add-DbaToolsToProfile
+    Updates your profile script for the current PowerShell host to import the
+    DbaTools module when the current PowerShell host starts.
+.EXAMPLE
+    PS C:\> Add-DbaToolsToProfile -AllHost
+    Updates your profile script for all PowerShell hosts to import the DbaTools
+    module whenever any PowerShell host starts.
+.INPUTS
+    None.
+.OUTPUTS
+    None.
+#>
+function Add-DbaToolsToProfile {
+    [CmdletBinding(SupportsShouldProcess)]
+    param(
+        [Parameter()]
+        [switch]
+        $AllHosts,
+
+        [Parameter()]
+        [switch]
+        $Force,
+
+        [Parameter()]
+        [switch]
+        $StartSshAgent,
+
+        [Parameter(ValueFromRemainingArguments)]
+        [psobject[]]
+        $TestParams
+    )
+
+    $underTest = $false
+
+    $profilePath = if ($AllHosts) { $PROFILE.CurrentUserAllHosts } else { $PROFILE.CurrentUserCurrentHost }
+
+    # Under test, we override some variables using $args as a backdoor.
+    if (($TestParams.Count -gt 0) -and ($TestParams[0] -is [string])) {
+        $profilePath = [string]$TestParams[0]
+        $underTest = $true
+        if ($TestParams.Count -gt 1) {
+            $ModuleBasePath = [string]$TestParams[1]
+        }
+    }
+
+    if (!$profilePath) { $profilePath = $PROFILE }
+
+    if (!$Force) {
+        # Search the user's profiles to see if any are using DbaTools already, there is an extra search
+        # ($profilePath) taking place to accomodate the Pester tests.
+        $importedInProfile = Test-DbaToolsImportedInScript $profilePath
+        if (!$importedInProfile -and !$underTest) {
+            $importedInProfile = Test-DbaToolsImportedInScript $PROFILE
+        }
+        if (!$importedInProfile -and !$underTest) {
+            $importedInProfile = Test-DbaToolsImportedInScript $PROFILE.CurrentUserCurrentHost
+        }
+        if (!$importedInProfile -and !$underTest) {
+            $importedInProfile = Test-DbaToolsImportedInScript $PROFILE.CurrentUserAllHosts
+        }
+        if (!$importedInProfile -and !$underTest) {
+            $importedInProfile = Test-DbaToolsImportedInScript $PROFILE.AllUsersCurrentHost
+        }
+        if (!$importedInProfile -and !$underTest) {
+            $importedInProfile = Test-DbaToolsImportedInScript $PROFILE.AllUsersAllHosts
+        }
+
+        if ($importedInProfile) {
+            Write-Warning "Skipping add of DbaTools import to file '$profilePath'."
```

Comment:
All of these (warnings/verbose/errors/host output) need to be adjusted to use the internal function `Write-Message`. You can see [the wiki](https://github.com/sqlcollaborative/dbatools/wiki/Implementing-the-messaging-system) for more info on how our messaging system works.